stuff/dataset_generator/runner.py

```diff
@@ -0,0 +1,188 @@
+"""
+Local request runner compatible with bootstrap signature.
+"""
+
+from __future__ import annotations
+
+import logging
+import time
+from typing import Any, Dict, Mapping, Optional, cast
+import yaml
+
+SETTINGS_PATH = "./stuff/dataset_generator/settings.yml"
+
+
+def _normalize_now(params: Dict[str, Any], now_ts: float) -> None:
+    """
+    replace $NOW$[+/-offset] tokens in-place with epoch seconds.
+    """
+    for key, val in list(params.items()):
+        if isinstance(val, str) and val.startswith("$NOW$"):
+            expr = val.replace("$NOW$", "", 1).strip()
+            offset = 0
+            if expr:
+                try:
+                    offset = int(expr)
+                except ValueError:
+                    offset = 0
+            params[key] = int(now_ts + offset)
+
+
+def _normalize_key(endpoint: str) -> str:
+    """
+    Normalize an endpoint for dictionary
+    lookups (lowercase + trailing slash).
+    """
+    normalized = endpoint.strip().lower()
+    return normalized if normalized.endswith("/") else normalized + "/"
+
+
+def _ensure_slash(endpoint: str) -> str:
+    """
+    Ensure trailing slash for real endpoints
+    while preserving case.
+    """
+    normalized = endpoint.strip()
+    return normalized if normalized.endswith("/") else normalized + "/"
+
+
+def _deep_get(
+    mapping: Mapping[str, object],
+    path: str
+) -> object | None:
+    """Return a nested value from a mapping using a dot-separated path."""
+    current: object = mapping
+    for key in path.split("."):
+        if not isinstance(current, Mapping) or key not in current:
+            return None
+        current = current[key]  # type: ignore[index]
```

Comment:
Hay forma de quitar este ignore? 
```suggestion
        current = current[key]
```
Lo mejor sería arreglar el error de raíz a querer ocultarlo